plugins/catalog-backend/src/ingestion/processors/GitLabDiscoveryProcessor.ts

```diff
@@ -94,7 +94,7 @@ export class GitLabDiscoveryProcessor implements CatalogProcessor {
     };
     for await (const project of projects) {
       result.scanned++;
-      if (!project.archived) {
+      if (!(project.archived || project.default_branch === undefined)) {
```

Comment:
we discussed in #8662 two ways to filter out "empty repo" projects, a) filter where `project.empty_repo === true`, b) filter where `project.default_branch === undefined`. as per @Rugvip's [suggestion](https://github.com/backstage/backstage/issues/8623#issuecomment-1001619984) we went with option b, as `project.empty_repo` isn't documented.

my followup here is: do we want to update this to `|| (branch === '*' && project.default_branch === undefined)`, so we only filter repos where default_branch is missing if a gitlab discovery is configured to scan projects for default branch? I just don't know enough about the gitlab api to know if there's a case where a repo has no default branch but _does_ have a branch you can manually read from. I doubt it but wanted to raise here just to flag

Reply:
Yep I think that extended check makes sense 👍